source/common/http/conn_manager_impl.cc

```diff
@@ -1118,8 +1120,17 @@ void ConnectionManagerImpl::ActiveStream::setBufferLimit(uint32_t new_limit) {
   }
 }
 
-void ConnectionManagerImpl::ActiveStream::createFilterChain() {
-  connection_manager_.config_.filterFactory().createFilterChain(*this);
+bool ConnectionManagerImpl::ActiveStream::createFilterChain() {
+  auto upgrade = request_headers_->Upgrade();
+  if (upgrade != nullptr) {
```

Comment:
What about the case where a user just wants to pass all upgrades to their upstream http server (because they want to deny upgrades, but not in a direct response from envoy)? I think we should always fallback to the non-upgrade filterChain if there is not a configured filter chain for this upgrade type.

Reply:
websocket is always point to point isn't it? even the current implementation is point to point to the extent that we upgrade and hope that the upstream will properly terminate the connection if it doesn't want to upgrade.

Reply:
I'm fine doing this for now, but note that for most early responses we no longer pass through the filter chain.  There's an open issue for responses to only pass through the filters which saw the request so I believe this behavior will not  persist in the long run.